test/userRegistrations.test.js

```diff
@@ -29,20 +29,65 @@ describe('User Registrations', ()=>{
   const findPendingUsersByEmail = async(email) => (await getPendingUsers()).filter(user => user.login_ids.find(k => k.id === email));
   const findPendingUsersByID = async(id) => (await getPendingUsers()).filter(user => user._id === id);
 
+  const FUNC_NAME = 'confirmFunc';
+  const FUNC_SOURCE = 'exports = function() { return { "status": "success" } }';
+  const createTestFunction = () => ({ name: FUNC_NAME, source: FUNC_SOURCE });
+
   it('removePendingUserByEmail should remove existing pending user', async() => {
     await client.register(testEmail, password);
     expect(await findPendingUsersByEmail(testEmail)).toHaveLength(1);
     await th.app().userRegistrations().removePendingUserByEmail(testEmail);
     expect(await findPendingUsersByEmail(testEmail)).toHaveLength(0);
-    expect(await getPendingUsers()).toHaveLength(1);
+    expect(await getPendingUsers()).toHaveLength(0);
   });
   it('removePendingUserByID should remove existing pending user', async() => {
     await client.register(testEmail, password);
     const userID = (await getPendingUsers())[0]._id;
     expect(await findPendingUsersByID(userID)).toHaveLength(1);
     await th.app().userRegistrations().removePendingUserByID(userID);
     expect(await findPendingUsersByID(userID)).toHaveLength(0);
-    expect(await getPendingUsers()).toHaveLength(1);
+    expect(await getPendingUsers()).toHaveLength(0);
+  });
+  it('rerunUserConfirmation should confirm existing pending user', async() => {
+    await client.register(testEmail, password);
+    expect(await findPendingUsersByEmail(testEmail)).toHaveLength(1);
+    let { token_id: tokenId, token } = await th.app().userRegistrations().rerunUserConfirmation(testEmail);
+    await client.auth.provider('userpass').emailConfirm(tokenId, token);
+
+    // We must login first so that a user id is attached to the password record
+    await client.login(testEmail, password);
+    expect(client.auth.loggedInProviderType).toEqual(PROVIDER_TYPE_USERPASS);
```

Comment:
Would this only be true if the email confirmation was successful?

Reply:
We wouldn't be able to login successfully if the email confirmation was not successful